337. House Robber III.py

```diff
@@ -0,0 +1,35 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+
+class Solution:
+    def rob(self, root):
+        """
+        :type root: TreeNode
+        :rtype: int
+        """
+        def loot(root, hm):
+            left, right, total = 0, 0, 0
+            
+            if not root:
+                return 0
+            elif root in hm:
+                return hm[root]
+            elif root:
```

Comment:
i think we can actually just put this entire else clause unindented. so
```
if not root:
...
elif root in hm:

[stuff]
```
Since that is the last clause.